### PR TITLE
Rails 5 support

### PR DESCRIPTION
**Pull request by benlovell**
Adds Rails 5 support. Everything was straight forward enough, except for some special consideration around f1b4ebb.

Best reviewed per commit. Ref #103.

cc @boffbowsh

### CODE REVIEW

File: app/controllers/api/user_controller.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 class Api::UserController < ActionController::Base
   include GDS::SSO::ControllerMethods
 
-  skip_before_filter :verify_authenticity_token
+  skip_before_filter :verify_authenticity_token, raise: false
```
Comment (boffbowsh) [inline app/controllers/api/user_controller.rb]: This is skipping a before filter, so surely it'd never raise / throw / return false anyway?

Comment (benlovell) [inline app/controllers/api/user_controller.rb]: You would think so, but this is the expected behaviour when the callback being skipped is yet to be defined (we end up in this code path): 

http://api.rubyonrails.org/classes/ActiveSupport/Callbacks/ClassMethods.html#method-i-skip_callback

Comment (boffbowsh) [inline app/controllers/api/user_controller.rb]: In that case, I don't think the commit message matches the problem. My understanding of it now is that if `verify_authenticity_token` isn't defined as a callback and you try to skip it, Rails will raise an error when it evaluates this line.

**Comment by boffbowsh** [comment]
Thanks for your work on this so far. I think we'll want to have a go at a solution to the capybara/mechanize issue before merging though. I'll try and find time v soon.

File: app/controllers/api/user_controller.rb (hunk 1 above)
Comment (benlovell) [inline app/controllers/api/user_controller.rb]: You're right. I've reworded that commit.

**Comment by boffbowsh** [comment]
Does https://github.com/alphagov/gds-sso/commit/c6ec519b0b3fa60eb4bfe0d564d04b9bdec458b3 address the problem do you think?

**Comment by benlovell** [comment]
@boffbowsh 👍 🚢

**Comment by boffbowsh** [comment]
Can you cherry-pick that onto your branch please and push it?

**Comment by benlovell** [comment]
@boffbowsh Done.

**Comment by boffbowsh** [comment]
The test failure is:
```
+ RBENV_VERSION=2.3
+ BUNDLE_PATH=/home/jenkins/bundles/govuk_gds_sso_branches
+ BUNDLE_GEMFILE=gemfiles/rails_4.2.gemfile
+ bundle exec rake
<snip>
Gem::LoadError: You have already activated bundler 1.10.6, but your Gemfile requires bundler 1.12.5. Prepending `bundle exec` to your command may solve this.
  /usr/lib/rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.12.5/lib/bundler/runtime.rb:35:in `block in setup'
  /usr/lib/rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.12.5/lib/bundler/runtime.rb:20:in `map'
  /usr/lib/rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.12.5/lib/bundler/runtime.rb:20:in `setup'
  /usr/lib/rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.12.5/lib/bundler.rb:95:in `setup'
  /usr/lib/rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.12.5/lib/bundler/setup.rb:19:in `<top (required)>'
  /usr/lib/rbenv/versions/2.2.4/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
  /usr/lib/rbenv/versions/2.2.4/lib/ruby/2.2.0/rubygems/core_ext/kernel_require.rb:54:in `require'
bundler: failed to load command: rake (/home/jenkins/workspace/govuk_gds_sso_branches/tmp/signonotron2_bundle/ruby/2.2.0/bin/rake)
```

I've had a quick look but can't make it fail locally, and can't see anything obvious. Full log is here: https://gist.github.com/boffbowsh/779b47950cac48912cd3ab91c4276cc9

**Comment by boffbowsh** [comment]
If you get no luck making it work, let me know and I'll find more time to look at it at some point.

**Comment by benlovell** [comment]
@boffbowsh Signon's Ruby is pinned at 2.2.3 and `BUNDLED_WITH` 1.10.6 which I assume is that Ruby's default bundler you folks have configured? I'm pretty sure bumping Signon's Ruby to 2.3.1 (and therefore its bundler) would resolve this. PR incoming.

**Comment by boffbowsh** [comment]
The signon branch is failing with the same error :(

**Comment by benlovell** [comment]
Hmm. I guess your 2.3.1 is hosed? Travelling for the next few days but I'll try and sneak a look if I get a moment.

**Comment by benlovell** [comment]
Rebased after #108 ❤️

**Comment by boffbowsh** [comment]
Needs another rebased and conflict resolution

**Comment by benlovell** [comment]
@boffbowsh Done.

**Pull request merged**